### DISCUSSION TITLE
Limit removal of Markers (points)

### DISCUSSION THREAD

**Discussion by cubecleveland** [geoman-io/leaflet-geoman#944] [Questions & Help]
Hello,

Is there any way to limit removal of a certain layer (namely markers) so the remove layer button will not actually remove markers.....
e.g removal will be enabled to all layers other than a pecific condition...

I was trying to listen to pm:remove and returning false but that didnt do the trick...

Thanks!

**Comment by Falke-Design** [comment]
You disable removal for all layers they have the PM-Option `allowRemoval: false`. Set this with `layer.pm.setOption({allowRemoval: false})`

**Reply by cubecleveland** [reply]
Thanks, I am wondering if i can disable removal of certain layers only.
E.g only limit removal of Markers.

**Reply by Falke-Design** [reply]
You can disable Marker removing after creation with this:

```
map.on('pm:create',(e)=>{
   if(e.shape === 'Marker'){
       e.layer.pm.setOption({allowRemoval: false})
   }
});
```
